tests/fastai/test_fastai_autolog.py

```diff
@@ -262,3 +264,34 @@ def test_fastai_autolog_non_early_stop_callback_does_not_log(fastai_random_data_
     # Check the test epoch numbers are correct
     assert num_of_epochs == NUM_EPOCHS
     assert len(metric_history) == num_of_epochs
+
+
+@pytest.mark.large
+@pytest.mark.parametrize("fit_variant", ["fit", "fit_one_cycle"])
+@pytest.mark.parametrize("callback", ["not-early"])
+@pytest.mark.parametrize("patience", [5])
+def test_fastai_autolog_batch_metrics_logger_logs_expected_metrics(fit_variant, callback, patience):
+    patched_metrics_data = []
+
+    # Mock patching BatchMetricsLogger.record_metrics()
+    # to ensure that expected metrics are being logged.
+    original = BatchMetricsLogger.record_metrics
+
+    with patch(
+        "mlflow.utils.autologging_utils.BatchMetricsLogger.record_metrics", autospec=True
+    ) as record_metrics_mock:
+
+        def record_metrics_side_effect(self, metrics, step=None):
+            patched_metrics_data.extend(metrics.items())
+            original(self, metrics, step)
+
+        record_metrics_mock.side_effect = record_metrics_side_effect
+        _, run = fastai_random_data_run_with_callback(
+            iris_data(), fit_variant, manual_run, callback, patience
+        )
+
+    patched_metrics_data = dict(patched_metrics_data)
+    original_metrics = run.data.metrics
```

Comment:
Can we add a small check to each of these test cases across the various modules?:

- Pick *one* concrete metric name (e.g., `train_loss`) and assert that the metric name is in `run.data.metrics`.

This assertion will guard against the case that our mocking logic somehow resulted in metrics being swallowed or inadvertently mutated. Looks like we do the this with Keras and TF Keras for  `restored_epoch` and `stopped_epoch`, which is awesome; let's add it to the other ones!